gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -303,6 +305,7 @@ public boolean apply(@Nullable WorkUnit workUnit) {
   public void launchJob(JobListener jobListener)
       throws JobException {
     String jobId = this.jobContext.getJobId();
+    MDC.put(ConfigurationKeys.JOB_KEY_KEY, this.jobContext.getJobKey());
```

Comment:
JOB_NAME should already be added by JobScheduler before calling `AbstractJobLauncher.launchJob`.